File: go/794_ValidTicTacToeState_test.go
package main

import (
    "github.com/stretchr/testify/assert"
    "testing"
)

func TestValidTicTacToe(t *testing.T) {
    assert.Equal(t,
        false,
        validTicTacToe([]string{"O  ", "   ", "   "}),
    )
    assert.Equal(t,
        false,
        validTicTacToe([]string{"XOX", " X ", "   "}),
    )
    assert.Equal(t,
        false,
        validTicTacToe([]string{"XXX", "   ", "OOO"}),
    )
    assert.Equal(t,
        true,
        validTicTacToe([]string{"XOX", "O O", "XOX"}),
    )
    assert.Equal(t,
        true,
        validTicTacToe([]string{"XXX", "OOX", "OOX"}),
    )
    assert.Equal(t,
        false,
        validTicTacToe([]string{"XXX", "XOO", "OO "}),
    )
}

File: python/794_ValidTicTacToeState_test.py
import unittest


class TestSolution(unittest.TestCase):

    def test_ValidTicTacToe(self):
        Solution = __import__('794_ValidTicTacToeState').Solution
        self.assertEqual(
            False,
            Solution().validTicTacToe(["O  ", "   ", "   "])
        )
        self.assertEqual(
            False,
            Solution().validTicTacToe(["XOX", " X ", "   "])
        )
        self.assertEqual(
            False,
            Solution().validTicTacToe(["XXX", "   ", "OOO"])
        )
        self.assertEqual(
            True,
            Solution().validTicTacToe(["XOX", "O O", "XOX"])
        )
        self.assertEqual(
            True,
            Solution().validTicTacToe(["XXX", "OOX", "OOX"])
        )
        self.assertEqual(
            False,
            Solution().validTicTacToe(["XXX", "XOO", "OO "])
        )
